src/services/codeAssignments.service.ts

```diff
@@ -0,0 +1,63 @@
+import { getRepository, IsNull } from 'typeorm'
+
+import { CodeAssignment } from 'devu-shared-modules'
+
+import CodeAssignmentsModel from '../model/codeAssignments.model'
+import { minioClient } from '../fileStorage'
+
+const connect = () => getRepository(CodeAssignmentsModel)
+
+function assignmentGraderFileRecordName(codeAssignment: CodeAssignment) {
+  if (!codeAssignment.id) throw new Error('Missing Id')
+  return codeAssignment.id.toString()
+}
+
+export async function create(codeAssignment: CodeAssignment, graderFile: Buffer) {
+  // I hope someone has a better idea for this function. The issue is with the grader file record naming.
+  // The name should depend on the codeAssignmentId to ensure there are no name conflicts, but we don't
+  // get that id until a record is created. So I create a record, grab the id and use it for the file record
+  // name, save the file, then update the record with the filename.
```

Comment:
We could have a guid column and use that instead? That way we could generate a unique id per codeAssignment, then create the file with that guid, then write that guid to the db.

Reply:
Do we want to lose meaningful file record names though?

Reply:
I'm not sure that using a guid vs a auto incrementing id makes it any more or less meaningful. 

Using the id would make it look way way way less obnoxious for sure though. Guids are kinda gross to look at.

Reply:
With I'd we keep the connection to a specific code assignment. I'm not sure what we want, but I thought this was a big talking point when we came up with the schema. If I'm miss remembering and it's not important to anyone then I don't care what we use. I think the argument was that in certain cases it's an extra db lookup to find a file

Reply:
I'd like to get input from @daviddob before deciding since this affects Leviathan